src/shared/codelens/codeLensUtils.ts

```diff
@@ -0,0 +1,183 @@
+import { SettingsConfiguration } from '../settingsConfiguration'
+
+// ------- Temporary logging utilities to be replaced by upcoming logger implementation --------------
+// TODO: Remove this code and replace usage with upcoming src/shared/logger
+type LogLevel = 'debug' | 'info' | 'warn' | 'error'
+const log = (channel: vscode.OutputChannel, level: LogLevel, msg: string, err?: Error): void => {
+    channel.show()
+    const _msg = `[${level}]: ${msg}`
+    if (err) {
+        // TODO: Define standard/strategy. Sometimes err.message is part of msg. Probably shouldn't be.
+        channel.appendLine(`${msg}: ${err.message}`)
+        console[level](new Date().toISOString(), _msg, err.stack)
+    } else {
+        channel.appendLine(msg)
+        console[level](new Date().toISOString(), _msg)
+    }
+}
+
+export enum OutputChannelName {
+    ToolKit = 'ToolKit',
+    Lambda = 'Lambda'
+}
+const outputChannels: {[channelName: string]: vscode.OutputChannel} = {
+    [OutputChannelName.ToolKit.toString()]: vscode.window.createOutputChannel(
+        localize('AWS.channel.aws.toolkit', 'AWS Toolkit')
+    ),
+    [OutputChannelName.Lambda.toString()]: vscode.window.createOutputChannel('AWS Lambda')
+}
+
+export const getOutputChannel = (name: OutputChannelName) => outputChannels[name.toString()]
+
+export const getLogger = (channelName: OutputChannelName) => {
+    const channel = outputChannels[channelName.toString()]
+
+    return {
+        debug: (msg: string) => log(channel, 'debug', msg),
+        info: (msg: string) => log(channel, 'info', msg),
+        warn: (msg: string) => log(channel, 'warn', msg),
+        error: (msg: string, err?: Error) => log(channel, 'error', msg, err)
+    }
+}
+
+const logger = getLogger(OutputChannelName.ToolKit)
+// ------- End temporary logging utilities to be replaced by upcoming logger implementation --------------
+
+export interface CodeLensProviderParams {
+    configuration: SettingsConfiguration,
+    toolkitOutputChannel: vscode.OutputChannel,
+    processInvoker?: SamCliProcessInvoker,
+    taskInvoker?: SamCliTaskInvoker
+}
+
+export const getLambdaHandlerCandidates = async ({uri}: {uri: vscode.Uri}): Promise<LambdaHandlerCandidate[]> => {
+    const filename = uri.fsPath
+
+    logger.info(`Getting symbols for '${uri.fsPath}'`)
+    const symbols: vscode.DocumentSymbol[] = ( // SymbolInformation has less detail (no children)
+        (await vscode.commands.executeCommand<vscode.DocumentSymbol[]>(
+            'vscode.executeDocumentSymbolProvider',
+            uri
+        )) || []
+    )
+
+    return symbols
+        .filter(sym => sym.kind === vscode.SymbolKind.Function)
+        .map(symbol => {
+            logger.debug(`Found potential handler: '${path.parse(filename).name}.${symbol.name}'`)
+
+            return {
+                filename,
+                handlerName: `${path.parse(filename).name}.${symbol.name}`,
+                positionStart: symbol.range.start.line,
```

Comment:
change to `symbol.range.start.character` (same for `end` on the next line)
https://code.visualstudio.com/api/references/vscode-api#Position

Reply:
I misunderstood Position.character (I thought it was document-indexed, not line-indexed).
In this case, the lines to assign to positionStart and positionEnd should be removed, since they are incorrect.